### DISCUSSION TITLE
Exception: list index out of range

### DISCUSSION THREAD

**Discussion by 2044745733** [langgenius/dify#11451] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Every time I perform a knowledge base search, the input question fails to be converted into a vector. I am using the shaw/dmeta-embedding-zh model on Ollama

### 2. Additional context or comments

model='shaw/dmeta-embedding-zh' embeddings=[] usage=EmbeddingUsage(tokens=4, total_tokens=4, unit_price=Decimal('0'), price_unit=Decimal('0'), total_price=Decimal('0E-7'), currency='USD', latency=0.05005987361073494)
2024-12-07 12:31:12.949 ERROR [Thread-772 (_generate_worker)] [node.py:74] - Node 1733563649719 failed to run
Traceback (most recent call last):
  File "/mnt/data8/liuhuan/dify-main/api/core/workflow/nodes/base/node.py", line 72, in run
    result = self._run()
             ^^^^^^^^^^^
  File "/mnt/data8/liuhuan/dify-main/api/core/workflow/nodes/knowledge_retrieval/knowledge_retrieval_node.py", line 65, in _run
    results = self._fetch_dataset_retriever(node_data=self.node_data, query=query)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/data8/liuhuan/dify-main/api/core/workflow/nodes/knowledge_retrieval/knowledge_retrieval_node.py", line 162, in _fetch_dataset_retriever
    all_documents = dataset_retrieval.multiple_retrieve(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/data8/liuhuan/dify-main/api/core/rag/retrieval/dataset_retrieval.py", line 420, in multiple_retrieve
    all_documents = data_post_processor.invoke(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/data8/liuhuan/dify-main/api/core/rag/data_post_processor/data_post_processor.py", line 37, in invoke
    documents = self.rerank_runner.run(query, documents, score_threshold, top_n, user)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/data8/liuhuan/dify-main/api/core/rag/rerank/weight_rerank.py", line 49, in run
    query_vector_scores = self._calculate_cosine(self.tenant_id, query, documents, self.weights.vector_setting)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/data8/liuhuan/dify-main/api/core/rag/rerank/weight_rerank.py", line 162, in _calculate_cosine
    query_vector = cache_embedding.embed_query(query)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/data8/liuhuan/dify-main/api/core/rag/embedding/cached_embedding.py", line 118, in embed_query
    raise ex
  File "/mnt/data8/liuhuan/dify-main/api/core/rag/embedding/cached_embedding.py", line 113, in embed_query
    embedding_results = embedding_result.embeddings[0]
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
![Snipaste_2024-12-07_20-58-43](https://github.com/user-attachments/assets/b51fdd81-7be4-401c-97e8-68d428710a0e)

**Comment by loganclark360** [comment]
> ### Self Checks
> 
> - [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
> - [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
> - [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
> - [X] Please do not modify this template :) and fill in all the required fields.
> 
> ### 1. Is this request related to a challenge you're experiencing? Tell me about your story.
> 
> Every time I perform a knowledge base search, the input question fails to be converted into a vector. I am using the shaw/dmeta-embedding-zh model on Ollama
> 
> ### 2. Additional context or comments
> 
> model='shaw/dmeta-embedding-zh' embeddings=[] usage=EmbeddingUsage(tokens=4, total_tokens=4, unit_price=Decimal('0'), price_unit=Decimal('0'), total_price=Decimal('0E-7'), currency='USD', latency=0.05005987361073494)
> 
> 2024-12-07 12:31:12.949 ERROR [Thread-772 (_generate_worker)] [node.py:74] - Node 1733563649719 failed to run
> 
> Traceback (most recent call last):
> 
>   File "/mnt/data8/liuhuan/dify-main/api/core/workflow/nodes/base/node.py", line 72, in run
> 
>     result = self._run()
> 
>              ^^^^^^^^^^^
> 
>   File "/mnt/data8/liuhuan/dify-main/api/core/workflow/nodes/knowledge_retrieval/knowledge_retrieval_node.py", line 65, in _run
> 
>     results = self._fetch_dataset_retriever(node_data=self.node_data, query=query)
> 
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "/mnt/data8/liuhuan/dify-main/api/core/workflow/nodes/knowledge_retrieval/knowledge_retrieval_node.py", line 162, in _fetch_dataset_retriever
> 
>     all_documents = dataset_retrieval.multiple_retrieve(
> 
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "/mnt/data8/liuhuan/dify-main/api/core/rag/retrieval/dataset_retrieval.py", line 420, in multiple_retrieve
> 
>     all_documents = data_post_processor.invoke(
> 
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "/mnt/data8/liuhuan/dify-main/api/core/rag/data_post_processor/data_post_processor.py", line 37, in invoke
> 
>     documents = self.rerank_runner.run(query, documents, score_threshold, top_n, user)
> 
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "/mnt/data8/liuhuan/dify-main/api/core/rag/rerank/weight_rerank.py", line 49, in run
> 
>     query_vector_scores = self._calculate_cosine(self.tenant_id, query, documents, self.weights.vector_setting)
> 
>                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "/mnt/data8/liuhuan/dify-main/api/core/rag/rerank/weight_rerank.py", line 162, in _calculate_cosine
> 
>     query_vector = cache_embedding.embed_query(query)
> 
>                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "/mnt/data8/liuhuan/dify-main/api/core/rag/embedding/cached_embedding.py", line 118, in embed_query
> 
>     raise ex
> 
>   File "/mnt/data8/liuhuan/dify-main/api/core/rag/embedding/cached_embedding.py", line 113, in embed_query
> 
>     embedding_results = embedding_result.embeddings[0]
> 
>                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
> 
> IndexError: list index out of range
> 
> ![Snipaste_2024-12-07_20-58-43](https://github.com/user-attachments/assets/b51fdd81-7be4-401c-97e8-68d428710a0e)
> 
>
